cli/server/websocketServer.ts

```diff
@@ -225,28 +254,40 @@ class WebSocketManager {
 				case 'delete-mapping': {
 					// Delete a mapping
 					const state = getServerState();
-					if (payload && payload.uuid) {
-						const uuid = payload.uuid as string;
-
-						const mapping = state.mappingsCache.get(uuid);
+					if (payload && payload.hash) {
+						const hash = payload.hash as string;
+
+						// Find the mapping by uuid in the cache values
+						let mapping: any = null;
+						let compositeKeyToDelete = '';
+						for (const [key, value] of state.mappingsCache.entries()) {
+							// the hash that is getting sent is the UUID
+
+							if (value.hash === hash) {
+								console.log(`Key ${key}\n Value hash: ${value.hash}\n Looking for hash: ${hash}`);
+								mapping = value;
+								compositeKeyToDelete = key;
+								break;
+							}
+						}
```

Comment:
Could just do a direct `state.mappingsCache.get()` here if we passed the composite key in rather than just the hash?